shuffle/cli/src/shared.rs

```diff
@@ -214,60 +218,30 @@ impl DevApiClient {
     }
 }
 
-// Contains all the commonly used paths in shuffle/cli
-pub struct Home {
-    account_path: PathBuf,
+pub struct NetworkHome {
+    shuffle_networks_path: PathBuf,
+    accounts_path: PathBuf,
+    latest_path: PathBuf,
     latest_address_path: PathBuf,
     latest_key_path: PathBuf,
-    latest_path: PathBuf,
-    networks_config_path: PathBuf,
-    node_config_path: PathBuf,
-    root_key_path: PathBuf,
-    shuffle_path: PathBuf,
-    test_key_address_path: PathBuf,
-    test_key_path: PathBuf,
     test_path: PathBuf,
-    validator_config_path: PathBuf,
-    validator_log_path: PathBuf,
+    test_key_path: PathBuf,
+    test_key_address_path: PathBuf,
 }
 
-impl Home {
-    pub fn new(home_path: &Path) -> Result<Self> {
-        Ok(Self {
-            account_path: home_path.join(".shuffle/accounts"),
-            latest_address_path: home_path.join(".shuffle/accounts/latest/address"),
-            latest_key_path: home_path.join(".shuffle/accounts/latest/dev.key"),
-            latest_path: home_path.join(".shuffle/accounts/latest"),
-            networks_config_path: home_path.join(".shuffle/Networks.toml"),
-            node_config_path: home_path.join(".shuffle/nodeconfig"),
-            root_key_path: home_path.join(".shuffle/nodeconfig/mint.key"),
-            shuffle_path: home_path.join(".shuffle"),
-            test_key_address_path: home_path.join(".shuffle/accounts/test/address"),
-            test_key_path: home_path.join(".shuffle/accounts/test/dev.key"),
-            test_path: home_path.join(".shuffle/accounts/test"),
-            validator_config_path: home_path.join(".shuffle/nodeconfig/0/node.yaml"),
-            validator_log_path: home_path.join(".shuffle/nodeconfig/validator.log"),
-        })
-    }
-
-    pub fn get_shuffle_path(&self) -> &Path {
-        &self.shuffle_path
-    }
-
-    pub fn get_root_key_path(&self) -> &Path {
-        &self.root_key_path
-    }
-
-    pub fn get_node_config_path(&self) -> &Path {
-        &self.node_config_path
-    }
-
-    pub fn get_validator_config_path(&self) -> &Path {
-        &self.validator_config_path
-    }
-
-    pub fn get_validator_log_path(&self) -> &Path {
-        &self.validator_log_path
+impl NetworkHome {
+    pub fn new(shuffle_networks_path: &Path, network_name: String) -> Self {
+        let network_path = shuffle_networks_path.join(network_name);
+        NetworkHome {
+            shuffle_networks_path: network_path.clone(),
```

Comment:
1. The passed in shuffle_networks_path is different to `NetworkHome#shuffle_networks_path` which to me is counterintuitive.
2. Should this be plural? Perhaps switch to network_base_path?

Reply:
instead of new taking in a path and network name, i'm just going to have it intake a network_base_path which will be `home.get_shuffle_path().join(network_name)`